sponsorship_compassion/models/res_partner.py

```diff
@@ -40,7 +40,8 @@ class ResPartner(models.Model):
     unrec_items = fields.Integer(compute='_compute_count_items')
     receivable_items = fields.Integer(compute='_compute_count_items')
     has_sponsorships = fields.Boolean()
-    number_sponsorships = fields.Integer()
+    number_sponsorships = fields.Integer(
+        compute='_compute_number_sponsorships')

```

Comment:
The field should be stored with store=True. The problem then is that then we don't know what we can put in @api.depends to recompute at the good time...